From d4daba5911f72be92e306985672455345cb6d612 Mon Sep 17 00:00:00 2001
From: avnerlam <avner.lamy@intel.com>
Date: Tue, 25 Apr 2017 11:41:26 +0300
Subject: [PATCH 636/639] PUMARTSYS-1101-Changes for L2VPN MAC-AGING

-Remove code under compilation flag CONFIG_INTEL_L2VPN_MAC_AGING (this flag is not set).
-Update PPSGC API header files with new APIs and types.

Signed-off-by: avnerlam <avner.lamy@intel.com>
---
 include/linux/avalanche/generic/avalanche_pp_api.h |   9 --
 include/linux/avalanche/puma7/ppsgc_db.h           |  11 ++
 include/linux/avalanche/puma7/ppsgc_defs.h         |   7 ++
 include/linux/ti_hil.h                             |  13 ---
 net/core/skbuff.c                                  |   7 --
 net/ti/pp/ti_hil_core.c                            | 114 ---------------------
 6 files changed, 18 insertions(+), 143 deletions(-)

diff --git a/include/linux/avalanche/generic/avalanche_pp_api.h b/include/linux/avalanche/generic/avalanche_pp_api.h
index 1cb9abe..67a4b28 100755
--- a/include/linux/avalanche/generic/avalanche_pp_api.h
+++ b/include/linux/avalanche/generic/avalanche_pp_api.h
@@ -1324,11 +1324,6 @@ typedef struct // former TI_PP_SESSION
 
     Uint8                   is_irreg_moca;
 #endif
-#ifdef CONFIG_INTEL_L2VPN_MAC_AGING
-    Uint8                   cpe_number;
-    Uint8                   spare_align_byte;
-    Uint16                  spare_align_word;
-#endif
 }
 AVALANCHE_PP_SESSION_INFO_t;
 
@@ -1713,10 +1708,6 @@ struct AVALANCHE_PP_SESSION_DELETE_CB_PARAM1_t{
 	Uint8	spare;
 };
 
-#ifdef CONFIG_INTEL_L2VPN_MAC_AGING
-#define AVALANCHE_PP_L2VPN_INVALID_CPE_NUMBER 0xff
-#endif
-
 #ifdef __KERNEL__
 /* **************************************************************************************** */
 /*                                                                                          */
diff --git a/include/linux/avalanche/puma7/ppsgc_db.h b/include/linux/avalanche/puma7/ppsgc_db.h
index 0a908b8..3eb6ede 100644
--- a/include/linux/avalanche/puma7/ppsgc_db.h
+++ b/include/linux/avalanche/puma7/ppsgc_db.h
@@ -125,6 +125,17 @@ int PPSGC_Db_DeleteGroupCounterSessions(unsigned short groupCounterId);
  */
 int PPSGC_Db_GetGroupCounterAmount(struct PpsgcGroupCounters *totalPpCounters);
 
+/*! \fn int PPSGC_Db_GetGroupSessionsInfo(unsigned int tableSize,
+ *              struct PpsgcGroupSessionsInfo *groupSessionsInfoTable)
+ *  \brief Retreives sessions information of sessions in specified
+ *         Group Counter table.
+ *  \param[in] tableSize - Group sessions info table size.
+ *  \param[in,out] groupSessionsInfoTable - Group sessions info table.
+ *  \return OK or error status.
+ */
+int PPSGC_Db_GetGroupSessionsInfo(unsigned int tableSize,
+		struct PpsgcGroupSessionsInfo *groupSessionsInfoTable);
+
 /*! \fn int PPSGC_Db_CreateSession
  *              (unsigned int sessionHandle, unsigned char sessionType,
  *               unsigned char multicast,    unsigned int ingressPortId,
diff --git a/include/linux/avalanche/puma7/ppsgc_defs.h b/include/linux/avalanche/puma7/ppsgc_defs.h
index f037001..00ce8eb 100644
--- a/include/linux/avalanche/puma7/ppsgc_defs.h
+++ b/include/linux/avalanche/puma7/ppsgc_defs.h
@@ -38,6 +38,7 @@
 #define PPSGC_GROUP_COUNTERS_PER_SESSION    (8)
 #define PPSGC_RC_FAILURE                    (-1)
 #define PPSGC_RC_SUCCESS                    (0)
+#define PPSGC_MAX_JIFFIES_VALUE             (0xFFFFFFFFFFFFFFFF)
 
 /*! \var struct PpsgcAssignedGroupCounters
     \brief Structure defines the assigned session group counter IDs.
@@ -65,6 +66,12 @@ struct PpsgcGroupCounters {
 	struct PpsgcPpCounters  ppCounters;
 };
 
+struct PpsgcGroupSessionsInfo {
+	unsigned short          groupCounterId;
+	unsigned short          sessionsNumber;
+	unsigned long long      lastSessionDelJiffies;
+};
+
 /*! \var struct PpsgcIfCounters
     \brief Structure defines the format of PP counters per interface device.
 */
diff --git a/include/linux/ti_hil.h b/include/linux/ti_hil.h
index e25a90c..6038cda 100755
--- a/include/linux/ti_hil.h
+++ b/include/linux/ti_hil.h
@@ -391,19 +391,6 @@ extern int ti_hil_register_delete_session_notification(TI_HIL_DELETE_SESSION ti_
 extern int ti_hil_unregister_delete_session_notification(void);
 #endif /* CONFIG_TI_PACKET_PROCESSOR_STATS */
 
-#ifdef CONFIG_INTEL_L2VPN_MAC_AGING
-/* DOCSIS Packet processor session notification API */
-typedef int (*TI_HIL_MAC_AGING_START_SESSION)(unsigned int sessionHandle, struct sk_buff* skb);
-typedef int (*TI_HIL_MAC_AGING_DELETE_SESSION)(unsigned char cpeNumber);
-
-extern int ti_hil_register_mac_aging_start_session_notification(
-	TI_HIL_MAC_AGING_START_SESSION ti_hil_start_session_notification);
-extern int ti_hil_unregister_mac_aging_start_session_notification(void);
-extern int ti_hil_register_mac_aging_delete_session_notification(
-	TI_HIL_MAC_AGING_DELETE_SESSION ti_hil_delete_session_notification);
-extern int ti_hil_unregister_mac_aging_delete_session_notification(void);
-#endif /* CONFIG_INTEL_L2VPN_MAC_AGING */
-
 #ifdef CONFIG_MACH_PUMA5
 /* Power Saving Mode (PSM) API */
 extern int ti_hil_enable_psm (void);
diff --git a/net/core/skbuff.c b/net/core/skbuff.c
index 1cbd397..a762fa1 100755
--- a/net/core/skbuff.c
+++ b/net/core/skbuff.c
@@ -299,13 +299,6 @@ int skb_init_intel(struct sk_buff *skb)
 #endif
 #endif  /* CONFIG_TI_PACKET_PROCESSOR */
 
-#ifdef CONFIG_TI_PACKET_PROCESSOR
-#ifdef CONFIG_INTEL_L2VPN_MAC_AGING
-	/* Indicating N.I.U cpe number */
-	SKB_GET_PP_INFO_P(skb)->pp_session.cpe_number =
-		AVALANCHE_PP_L2VPN_INVALID_CPE_NUMBER;
-#endif
-#endif
     return 0;
 }
 
diff --git a/net/ti/pp/ti_hil_core.c b/net/ti/pp/ti_hil_core.c
index 3c61924..e7d0c5f 100755
--- a/net/ti/pp/ti_hil_core.c
+++ b/net/ti/pp/ti_hil_core.c
@@ -170,15 +170,6 @@ TI_HIL_DELETE_SESSION ti_hil_delete_session_notification_cb = NULL;
 EXPORT_SYMBOL(ti_hil_delete_session_notification_cb);
 #endif /* CONFIG_TI_PACKET_PROCESSOR_STATS */
 
-#ifdef CONFIG_INTEL_L2VPN_MAC_AGING
-/* DOCSIS Packet processor MAC AGING start session notification Callback */
-TI_HIL_MAC_AGING_START_SESSION ti_hil_mac_aging_start_session_notification_cb = NULL;
-EXPORT_SYMBOL(ti_hil_mac_aging_start_session_notification_cb);
-/* DOCSIS Packet processor MAC AGING delete session notification Callback */
-TI_HIL_MAC_AGING_DELETE_SESSION ti_hil_mac_aging_delete_session_notification_cb = NULL;
-EXPORT_SYMBOL(ti_hil_mac_aging_delete_session_notification_cb);
-#endif
-
 /**************************************************************************
  ******************************* Functions  *******************************
  **************************************************************************/
@@ -656,104 +647,6 @@ int ti_hil_unregister_delete_session_notification(void)
 }
 #endif /* CONFIG_TI_PACKET_PROCESSOR_STATS */
 
-#ifdef CONFIG_INTEL_L2VPN_MAC_AGING
-/**************************************************************************
- * FUNCTION NAME : ti_hil_register_mac_aging_start_session_notification
- **************************************************************************
- * DESCRIPTION   :
- *  Register the DOCSIS Packet Processor start session notification for mac aging feature
- *  callback function
- *
- * RETURNS       :
- *  0   -   Success
- *  <0  -   Error
- **************************************************************************/
-int ti_hil_register_mac_aging_start_session_notification(
-	TI_HIL_MAC_AGING_START_SESSION ti_hil_start_session_notification)
-{
-	if (ti_hil_mac_aging_start_session_notification_cb != NULL) {
-		pr_err("Error: DOCSIS Packet Processor mac aging start session notification callback already exist\n");
-		return -1;
-	}
-
-	ti_hil_mac_aging_start_session_notification_cb =
-		ti_hil_start_session_notification;
-
-	return 0;
-}
-
-/**************************************************************************
- * FUNCTION NAME : ti_hil_unregister_mac_aging_start_session_notification
- **************************************************************************
- * DESCRIPTION   :
- *  Un-Register the DOCSIS Packet Processor start session notification for mac aging feature
- *  callback function
- *
- * RETURNS       :
- *  0   -   Success
- *  <0  -   Error
- **************************************************************************/
-int ti_hil_unregister_mac_aging_start_session_notification(void)
-{
-	if (ti_hil_mac_aging_start_session_notification_cb == NULL) {
-		pr_err("Error: DOCSIS Packet Processor mac aging start session notification callback empty\n");
-		return -1;
-	}
-
-	ti_hil_mac_aging_start_session_notification_cb = NULL;
-
-	return 0;
-}
-
-/**************************************************************************
- * FUNCTION NAME : ti_hil_register_mac_aging_delete_session_notification
- **************************************************************************
- * DESCRIPTION   :
- *  Register the DOCSIS Packet processor delete session notification for mac aging feature
- *  callback function
- *
- * RETURNS       :
- *  0   -   Success
- *  <0  -   Error
- **************************************************************************/
-int ti_hil_register_mac_aging_delete_session_notification(
-	TI_HIL_MAC_AGING_DELETE_SESSION ti_hil_delete_session_notification)
-{
-	if (ti_hil_mac_aging_delete_session_notification_cb != NULL) {
-		pr_err("Error: DOCSIS Packet Processor delete session notification callback already exist\n");
-		return -1;
-	}
-
-	ti_hil_mac_aging_delete_session_notification_cb =
-		ti_hil_delete_session_notification;
-
-	return 0;
-}
-
-/**************************************************************************
- * FUNCTION NAME : ti_hil_unregister_mac_aging_delete_session_notification
- **************************************************************************
- * DESCRIPTION   :
- *  Un-Register the DOCSIS Packet processor delete session notification for mac aging feature
- *  callback function
- *
- * RETURNS       :
- *  0   -   Success
- *  <0  -   Error
- **************************************************************************/
-int ti_hil_unregister_mac_aging_delete_session_notification(void)
-{
-	if (ti_hil_mac_aging_delete_session_notification_cb == NULL) {
-		pr_err("Error: DOCSIS Packet Processor mac aging delete session notification callback empty\n");
-		return -1;
-	}
-
-	ti_hil_mac_aging_delete_session_notification_cb = NULL;
-
-	return 0;
-}
-#endif CONFIG_INTEL_L2VPN_MAC_AGING
-
 /* Export all the Symbols for Linux; so that these can be called from modules. */
 EXPORT_SYMBOL(ti_hil_initialize);
 EXPORT_SYMBOL(ti_hil_register_profile);
@@ -765,11 +658,4 @@ EXPORT_SYMBOL(ti_hil_unregister_start_session_notification);
 EXPORT_SYMBOL(ti_hil_register_delete_session_notification);
 EXPORT_SYMBOL(ti_hil_unregister_delete_session_notification);
 #endif /* CONFIG_TI_PACKET_PROCESSOR_STATS */
-#ifdef CONFIG_INTEL_L2VPN_MAC_AGING
-/* DOCSIS Packet processor start/delete session notification API */
-EXPORT_SYMBOL(ti_hil_register_mac_aging_start_session_notification);
-EXPORT_SYMBOL(ti_hil_unregister_mac_aging_start_session_notification);
-EXPORT_SYMBOL(ti_hil_register_mac_aging_delete_session_notification);
-EXPORT_SYMBOL(ti_hil_unregister_mac_aging_delete_session_notification);
-#endif /* CONFIG_INTEL_L2VPN_MAC_AGING */
 
-- 
2.10.1

